Kuadrant/multicluster-gateway-controller
# Additional headers as secret

Change additional headers to a secret ref, so that users can store auth keys, etc. in it.

Update DNS Policy document with new fields. 

## verification
- create a workload and see DNSRecords
- Create a secret with a key / value pair, where the value ends with "\n" (e.g. "test\n": `dGVzdAo=`)
- Create a health check spec that references this secret
- See the health check failing due to the invalid header (may take upto a minute)
- Change the secret value to no longer end with "\n" (e.g. "test": `dGVzdA==`)
- See the health check begin passing (may take upto a minute)
